x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IlmHealthIndicatorService.java

```diff
@@ -118,21 +121,21 @@ public class IlmHealthIndicatorService implements HealthIndicatorService {
         SearchableSnapshotAction.NAME,
         actionRule(SearchableSnapshotAction.NAME).maxTimeOnAction(ONE_DAY)
             .stepRules(
-                stepRule(WaitForDataTierStep.NAME, ONE_DAY),
-                stepRule(WaitForIndexColorStep.NAME, ONE_DAY),
+                stepRuleFullChecks(WaitForDataTierStep.NAME, ONE_DAY, MAX_RETRIES),
+                stepRuleFullChecks(WaitForIndexColorStep.NAME, ONE_DAY, MAX_RETRIES),
                 // The no-follower step is added here because an `UnfollowAction` is added before the `shrinkAction` in the follower cluster
-                stepRule(WaitForNoFollowersStep.NAME, ONE_DAY)
+                stepRuleOnlyCheckRetries(WaitForNoFollowersStep.NAME, MAX_RETRIES)
             ),
         //
         DeleteAction.NAME,
-        actionRule(DeleteAction.NAME).stepRules(stepRule(DeleteStep.NAME, ONE_DAY)),
+        actionRule(DeleteAction.NAME).stepRules(stepRuleFullChecks(DeleteStep.NAME, ONE_DAY, MAX_RETRIES)),
         //
         ShrinkAction.NAME,
         actionRule(ShrinkAction.NAME).maxTimeOnAction(ONE_DAY)
             .stepRules(
                 // The no-follower step is added here because an `unfollowAction` is added before the `shrinkAction` in the follower
                 // cluster.
-                stepRule(WaitForNoFollowersStep.NAME, ONE_DAY)
+                stepRuleOnlyCheckRetries(WaitForNoFollowersStep.NAME, MAX_RETRIES)
```

Comment:
The comment can be removed - now that we're only looking at max retries for the `WaitForNoFollowerStep`.
(same for above in the `SearchableSnapshotAction`) 

And thinking about it some more, I think it's ok to **not** have `MAX_TIME` on this step because we can have policies without rollover. So a policy with only the `warm` phase and a `shrink` action could stay in `WaitForNoFollowerStep` on the leader cluster for a long time if the index is actively written to still or if a user on the follower cluster doesn't **manually** (this is a policy without rollover - the rollover action sets this automatically otherwise) set the `indexing_complete` setting to true